eis_qgis_plugin/processing/algorithms/validation/plot_correlation_matrix.py

```diff
@@ -0,0 +1,64 @@
+from qgis.core import (
+    QgsProcessingParameterEnum,
+    QgsProcessingParameterFile,
+    QgsProcessingParameterFileDestination,
+    QgsProcessingParameterBoolean,
+    QgsProcessingParameterString,
+)
+
+from eis_qgis_plugin.processing.eis_processing_algorithm import EISProcessingAlgorithm
+
+
+class EISPlotCorrelationMatrix(EISProcessingAlgorithm):
+    def __init__(self) -> None:
+        super().__init__()
+
+        self._name = "plot_correlation_matrix"
+        self._display_name = "Plot correlation matrix"
+        self._group = "Validation"
+        self._group_id = "validation"
+        self._short_help_string = "Create a heatmap to visualize correlation matrix."
+
+    def initAlgorithm(self, config=None):
+        self.alg_parameters = ["matrix", "annotate", "cmap", "plot_title", "**kwargs", "out_raster"]
+
+        self.addParameter(
+            QgsProcessingParameterFile(
+                name=self.alg_parameters[0],
+                description="Correlation matrix.",
+            )
+        )
```

Comment:
I'm not 100% sure, but I think this function is used for vector data, so you could change the input to that (this function call would actually both produce the correlation matrix and then plot it).